client/browser/src/libs/options/ServerURLForm.tsx

```
@@ -137,6 +139,15 @@ export class ServerURLForm extends React.Component<ServerURLFormProps> {
                             </a>
                             .
                         </p>
+                        {!this.props.urlHasPermissions && (
+                            <p>
+                                You may need to{' '}
+                                <a href="#" onClick={this.props.requestPermissions}>
```

Use `href=""`, otherwise the URL will be modified and the page can scroll to the top.

---

Can the Promise reject? If yes, that error is not handled

---

`href=""` causes the options page to reload. I think `href="#"` is fine here, as the options page is rendered as a popup, not in-page, and has no scroll. It will also be closed as soon as it loses focus, in this case when the user interacts with the permissions prompt.

The promise [cannot reject](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/browser/src/browser/permissions.ts#L9-21).